tests/rest/client/test_third_party_rules.py

```diff
@@ -161,6 +161,68 @@ async def check(ev: EventBase, state):
         ev = channel.json_body
         self.assertEqual(ev["content"]["x"], "y")
 
+    def test_message_edit(self):
+        """Ensure that the module doesn't cause issues with edited messages."""
+        # first patch the event checker so that it will modify the event
+        async def check(ev: EventBase, state):
+            d = ev.get_dict()
+            d["content"] = {
+                "msgtype": "m.text",
+                "body": d["content"]["body"].upper(),
+            }
+            return d
```

Comment:
I'm a bit confused as we try to prevent modules from modifying events? https://github.com/matrix-org/synapse/blob/0a00b7ff14890987f09112a2ae696c61001e6cf1/synapse/events/third_party_rules.py#L73-L76

Reply:
How is this different than the test above that does the same thing? 😕 Maybe I misunderstood what that test was doing...

Reply:
Ahaaaaa. It was me who was misunderstanding what the point of freezing the event was.

So we do allow `ThirdPartyRules` modules to modify event dicts, but it must do so by returning a dict with the new fields, and must not modify the original event object.

So the freeze is just to ensure that last bit - but we still allow modifying the event.

In that case yes, this test looks solid!